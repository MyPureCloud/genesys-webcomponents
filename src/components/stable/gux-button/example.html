<h4>Enabled</h4>
<div>
  <gux-button title="default">Default</gux-button>
  <gux-button title="Primary" accent="primary">Primary</gux-button>
  <gux-button title="Secondary" accent="secondary">Secondary</gux-button>
  <gux-button title="Invalid accent" accent="invalid"
    >Invalid accent</gux-button
  >
  <gux-button title="Slotted element" accent="invalid">
    <span>Text in a span element</span>
  </gux-button>
  <gux-button title="Slotted element" accent="invalid">
    <span>Text in a span element</span>
    <span>More text in a span element</span>
  </gux-button>
</div>

<h4>Disabled</h4>
<div>
  <gux-button title="default" disabled>Default</gux-button>
  <gux-button title="Primary" accent="primary" disabled>Primary</gux-button>
  <gux-button title="Secondary" accent="secondary" disabled
    >Secondary</gux-button
  >
  <gux-button title="Invalid accent" accent="invalid" disabled
    >Invalid accent</gux-button
  >
  <gux-button title="Slotted element" accent="invalid" disabled>
    <span>Text in a span element</span>
  </gux-button>
  <gux-button title="Slotted element" accent="invalid" disabled>
    <span>Text in a span element</span>
    <span>More text in a span element</span>
  </gux-button>
</div>

<style>
  gux-button {
    margin: 10px;
  }
</style>
