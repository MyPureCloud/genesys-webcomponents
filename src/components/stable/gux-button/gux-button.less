@import (reference) '../../../style/color.less';

// Variables part
@primary-background-color: @gux-genesys-dark-blue;
@primary-border-color: #2455a3;
@primary-color: @gux-type-dark;
@primary-active-background-color: #275aad;

@primary-dark-background-color: @gux-genesys-dark-blue;
@primary-dark-border-color: @gux-grey-global;
@primary-dark-color: @gux-type-dark;
@primary-dark-hover-background-color: #3476f5;

@secondary-background-color: @gux-border;
@secondary-border-color: #d5dce0;
@secondary-color: @gux-type;
@secondary-active-background-color: #dce1e5;

@secondary-dark-background-color: #555d66;
@secondary-dark-border-color: @gux-grey-global;
@secondary-dark-color: @gux-type-dark;
@secondary-dark-hover-background-color: #76818f;

gux-button {
  display: inline-block;
  min-width: 32px;
  pointer-events: none;

  > button {
    width: 100%;
    padding: 0 16px;
    height: 32px;
    border-radius: 2px;
    cursor: pointer;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    font-size: 12px;
    font-weight: bold;
    pointer-events: auto;

    &[disabled] {
      opacity: 0.5;
      cursor: default;
    }

    &:focus {
      outline: none;
    }
  }
}

// Theming
.gux-button-dark-theme {
  > button {
    &.gux-secondary {
      background-color: @secondary-dark-background-color;
      border: 1px solid @secondary-dark-border-color;
      color: @secondary-dark-color;

      &:hover {
        background-color: @secondary-dark-hover-background-color;
      }

      &:active {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 1);
      }

      &:focus {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 0.5);
      }

      &[disabled] {
        &:hover {
          background-color: @secondary-dark-background-color;
        }

        &:active {
          box-shadow: none;
        }

        &:focus {
          box-shadow: none;
        }
      }
    }

    &.gux-primary {
      background-color: @primary-dark-background-color;
      border: 1px solid @primary-dark-border-color;
      color: @primary-dark-color;

      &:hover {
        background-color: @primary-dark-hover-background-color;
      }

      &:active {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 1);
      }

      &:focus {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 0.5);
      }

      &[disabled] {
        &:hover {
          background-color: @primary-dark-background-color;
        }

        &:active {
          box-shadow: none;
        }

        &:focus {
          box-shadow: none;
        }
      }
    }
  }
}

.gux-button-light-theme {
  > button {
    &.gux-secondary {
      background-color: @secondary-background-color;
      border: 1px solid @secondary-border-color;
      color: @secondary-color;

      &:hover {
        box-shadow: 0 0 1px 1px rgba(34, 37, 41, 0.24);
      }

      &:active {
        background-color: @secondary-active-background-color;
      }

      &:focus {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 0.5);
      }

      &[disabled] {
        &:hover {
          box-shadow: none;
        }

        &:active {
          background-color: @secondary-background-color;
        }

        &:focus {
          box-shadow: none;
        }
      }
    }

    &.gux-primary {
      background-color: @primary-background-color;
      border: 1px solid @primary-border-color;
      color: @primary-color;

      &:hover {
        box-shadow: 0 0 1px 1px rgba(34, 37, 41, 0.48);
      }

      &:active {
        background-color: @primary-active-background-color;
      }

      &:focus {
        box-shadow: 0 0 0 3px rgba(117, 168, 255, 0.5);
      }

      &[disabled] {
        &:hover {
          box-shadow: none;
        }

        &:active {
          background-color: @primary-background-color;
        }

        &:focus {
          box-shadow: none;
        }
      }
    }
  }
}

// Default Theme
gux-button {
  .gux-button-light-theme();
}

// Dark
.gux-dark-theme {
  gux-button {
    .gux-button-dark-theme();
  }
}

gux-button.gux-dark-theme {
  .gux-button-dark-theme();
}

// Light
.gux-light-theme {
  gux-button {
    .gux-button-light-theme();
  }
}

gux-button.gux-light-theme {
  .gux-button-light-theme();
}
