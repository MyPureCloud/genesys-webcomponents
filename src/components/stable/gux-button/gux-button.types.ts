export type GuxButtonAccent = 'primary' | 'secondary';

export type GuxButtonType = 'button' | 'submit' | 'reset';
