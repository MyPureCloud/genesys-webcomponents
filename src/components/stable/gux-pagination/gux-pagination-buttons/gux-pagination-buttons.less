gux-pagination-buttons {
  .gux-pagination-buttons-container {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: flex-end;
    align-content: stretch;
    align-items: center;

    &.expanded {
      justify-content: center;
    }

    & > div {
      order: 0;
      flex: 0 1 auto;
      align-self: auto;
    }

    .gux-pagination-buttons-group {
      white-space: nowrap;

      gux-button:not(:first-of-type) {
        margin-left: 4px;
      }
    }

    .gux-pagination-buttons-input-container {
      margin: 0 16px;
      display: flex;
      flex-direction: row;
      flex-wrap: nowrap;
      justify-content: center;
      align-content: stretch;
      align-items: center;

      & > div {
        order: 0;
        flex: 0 1 auto;
        align-self: auto;
        white-space: nowrap;
      }

      .gux-pagination-buttons-input {
        width: 60px;
        margin: 0 8px;

        input {
          text-align: center;
        }
      }
    }

    .gux-pagination-buttons-list-container {
      margin: 0 16px;
      display: flex;
      flex-direction: row;
      flex-wrap: nowrap;
      justify-content: center;
      align-content: stretch;
      align-items: center;

      .gux-pagination-buttons-list-current,
      .gux-pagination-buttons-list-target {
        margin: 0;
        padding: 0 8px;
        border: none;
        background: inherit;
        height: 32px;
      }

      .gux-pagination-buttons-list-current {
        color: @gux-grey-2;
        font-weight: 600;
      }

      .gux-pagination-buttons-list-target {
        color: @gux-dark-blue;
        cursor: pointer;
      }
    }

    .gux-pagination-buttons-spacer {
      width: 10px;
    }
  }
}

// Theming

// Dark
.gux-pagination-buttons-dark-theme {
  color: @gux-off-white;
};

.gux-dark-theme {
  gux-pagination-buttons {
      .gux-pagination-buttons-dark-theme();
  }
}

gux-pagination-buttons.gux-dark-theme {
  .gux-pagination-buttons-dark-theme();
}

// Light
.gux-pagination-buttons-light-theme {
  color: @gux-grey-2;
};

.gux-light-theme {
  gux-pagination-buttons {
      .gux-pagination-buttons-light-theme();
  }
}

gux-pagination-buttons.gux-light-theme {
  .gux-pagination-buttons-light-theme();
}

// Default Theme
gux-pagination-buttons {
  .gux-pagination-buttons-light-theme();
}
