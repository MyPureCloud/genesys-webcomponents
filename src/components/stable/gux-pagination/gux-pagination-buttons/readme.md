# gux-pagination-buttons

An internal component used by the gux-pagination component.


<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description | Type                              | Default     |
| ------------- | -------------- | ----------- | --------------------------------- | ----------- |
| `currentPage` | `current-page` |             | `number`                          | `undefined` |
| `layout`      | `layout`       |             | `"expanded" \| "full" \| "small"` | `'full'`    |
| `totalPages`  | `total-pages`  |             | `number`                          | `undefined` |


## Events

| Event                       | Description | Type                  |
| --------------------------- | ----------- | --------------------- |
| `internalcurrentpagechange` |             | `CustomEvent<number>` |


## Dependencies

### Used by

 - [gux-pagination](..)

### Depends on

- [gux-input-text-like](../../gux-form-field/components/gux-input-text-like)
- [gux-button](../../gux-button)
- [gux-icon](../../gux-icon)

### Graph
```mermaid
graph TD;
  gux-pagination-buttons --> gux-input-text-like
  gux-pagination-buttons --> gux-button
  gux-pagination-buttons --> gux-icon
  gux-pagination --> gux-pagination-buttons
  style gux-pagination-buttons fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
