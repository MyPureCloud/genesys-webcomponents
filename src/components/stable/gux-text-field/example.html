<gux-text-field></gux-text-field>
<gux-text-field value="Value"></gux-text-field>
<gux-text-field
  value="Lorem ipsum dolor sit amet, lacinia blandit ultricies lorem aliquam, dui quam a viverra vivamus pellentesque, rhoncus aliquet proin eleifend."
></gux-text-field>
