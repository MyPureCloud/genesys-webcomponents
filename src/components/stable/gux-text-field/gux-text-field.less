// Variables part
@import (reference) '../../../style/shared/text-input.less';
@import (reference) '../../../style/color.less';

@background-color: @gux-type-dark;
@text-color: @gux-type;
@border: #dae1e8;
@placeholder: #9baaba;
@icon-color: #b1becc;

@dark-background-color: @gux-type;
@dark-text-color: @gux-type-dark;
@dark-border: @gux-grey-global;
@dark-placeholder: #9baaba;
@dark-icon-color: #b1becc;

@focus-outline: fade(@gux-genesys-blue, 50%);
@focus-border: @gux-genesys-dark-blue;
@alert-color: @gux-alert;
@alert-color-dark: @gux-alert-dark;

gux-text-field {
  display: block;

  > div {
    margin: 4px 0;

    &.disabled {
      opacity: 0.5;
      pointer-events: none;
      cursor: default;
    }

    &.error,
    &.warning {
      input {
        border-color: @alert-color;
        outline: 0;

        &:focus,
        &:hover {
          border-color: @alert-color;
          outline: 0;
        }
      }

      .gux-error {
        color: @alert-color;

        i {
          margin-right: 4px;
        }
      }
    }

    &.warning {
      .gux-error {
        i {
          color: @gux-warning;
        }
      }
    }

    > div.gux-field {
      position: relative;
      height: 32px;
    }
  }

  .gux-field {
    position: relative;

    input {
      .input-style();

      &.text-clearable {
        padding-right: 25px;
      }
    }

    button {
      color: @icon-color;
      position: absolute;
      top: 1px;
      right: 3px;
      bottom: 0;
      border: none;
      background: none;
      cursor: pointer;
      display: block;
      outline: none;
      overflow: hidden;
    }
  }
}

// Theming

// Dark
.gux-text-field-dark-theme {
  > div {
    &.error,
    &.warning {
      input {
        border-color: @alert-color-dark;
        outline: 0;

        &:focus,
        &:hover {
          border-color: @alert-color-dark;
          outline: 0;
        }
      }

      .gux-error {
        color: @alert-color-dark;

        i {
          margin-right: 4px;
        }
      }
    }

    &.warning {
      .gux-error {
        i {
          color: @gux-warning;
        }
      }
    }
  }

  .gux-field {
    input {
      .input-dark-theme();
    }

    button {
      color: @dark-icon-color;
    }
  }

  &:not([disabled]):hover,
  &:not([disabled]):focus {
    button {
      opacity: 1;
    }
  }
}

.gux-dark-theme {
  gux-text-field {
    .gux-text-field-dark-theme();
  }
}

.gux-text-field.gux-dark-theme {
  .gux-text-field-dark-theme();
}

// Light
.gux-text-field-light-theme {
  > div {
    &.error,
    &.warning {
      input {
        border-color: @alert-color;

        &:focus,
        &:hover {
          border-color: @alert-color;
        }
      }

      .gux-error {
        color: @alert-color;
      }
    }

    &.warning {
      .gux-error {
        i {
          color: @gux-warning;
        }
      }
    }
  }

  .gux-field {
    input {
      background-color: @background-color;
      border: 1px solid @border;
      color: @text-color;

      &::placeholder {
        color: @placeholder;
      }

      &::-webkit-input-placeholder {
        color: @placeholder;
      }

      &::-ms-input-placeholder {
        color: @placeholder;
      }

      &::-moz-placeholder {
        color: @placeholder;
      }

      &:focus {
        border: 1px solid @focus-border;
        box-shadow: 0 0 4px @focus-outline;
      }
    }

    button {
      color: @icon-color;
    }
  }
}

.gux-light-theme {
  gux-text-field {
    .gux-text-field-light-theme();
  }
}

gux-text-field.gux-light-theme {
  .gux-text-field-light-theme();
}

// Default Theme
gux-text-field {
  .gux-text-field-light-theme();
}
