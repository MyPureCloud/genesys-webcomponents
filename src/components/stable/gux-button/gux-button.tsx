import { Component, Element, h, JSX, Prop } from '@stencil/core';

import { GuxButtonAccent, GuxButtonType } from './gux-button.types';

@Component({
  styleUrl: 'gux-button.less',
  tag: 'gux-button'
})
export class GuxButton {
  @Element()
  private root: HTMLElement;

  /**
   * The component button type
   */
  @Prop()
  type: GuxButtonType = 'button';

  /**
   * The component title
   */
  @Prop()
  title: string;

  /**
   * Indicate if the button is disabled or not
   */
  @Prop()
  disabled = false;

  /**
   * The component accent (secondary or primary).
   */
  @Prop()
  accent: GuxButtonAccent = 'secondary';

  componentWillLoad() {
    this.makeSlotContentDisableable();
  }

  render(): JSX.Element {
    return (
      <button
        type={this.type}
        title={this.title}
        disabled={this.disabled}
        class={this.accent === 'primary' ? 'gux-primary' : 'gux-secondary'}
      >
        <slot />
      </button>
    );
  }

  private makeSlotContentDisableable() {
    Array.from(this.root.children).forEach(slotElement => {
      slotElement.addEventListener('click', (event: MouseEvent): void => {
        if (this.disabled) {
          event.stopImmediatePropagation();
          event.stopPropagation();
          event.preventDefault();
        }
      });
    });
  }
}
