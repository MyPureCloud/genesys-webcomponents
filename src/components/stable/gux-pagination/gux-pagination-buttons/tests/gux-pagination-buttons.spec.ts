import { newSpecPage } from '@stencil/core/testing';
import { GuxPaginationButtons } from '../gux-pagination-buttons';
import { GuxButton } from '../../../gux-button/gux-button';
import { GuxTextFieldLegacy } from '../../../../legacy/gux-text-field-legacy/gux-text-field';

const components = [GuxPaginationButtons, GuxButton, GuxTextFieldLegacy];
const language = 'en';

describe('gux-pagination-item-counts', () => {
  describe('#render', () => {
    [
      { totalPages: 0, currentPage: 0, layout: 'full' },
      { totalPages: 10, currentPage: 1, layout: 'full' },
      { totalPages: 10, currentPage: 5, layout: 'full' },
      { totalPages: 0, currentPage: 0, layout: 'small' },
      { totalPages: 10, currentPage: 1, layout: 'small' },
      { totalPages: 10, currentPage: 5, layout: 'small' },
      { totalPages: 0, currentPage: 0, layout: 'expanded' },
      { totalPages: 10, currentPage: 1, layout: 'expanded' },
      { totalPages: 10, currentPage: 5, layout: 'expanded' }
    ].forEach(({ totalPages, currentPage, layout }, index) => {
      it(`should render as expected (${index + 1})`, async () => {
        const html = `
          <gux-pagination-buttons
            total-pages="${totalPages}"
            current-page="${currentPage}"
            layout="${layout}"
          >
          </gux-pagination-buttons>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxPaginationButtons);

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
