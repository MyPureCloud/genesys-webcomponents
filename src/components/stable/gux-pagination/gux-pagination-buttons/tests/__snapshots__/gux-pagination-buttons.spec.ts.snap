// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-pagination-item-counts #render should render as expected (1) 1`] = `
<gux-pagination-buttons current-page="0" layout="full" total-pages="0">
  <div class="gux-full gux-pagination-buttons-container">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-input-container">
      <div>
        Page
      </div>
      <div class="gux-pagination-buttons-input">
        <gux-text-field use-clear-button="false" value="0">
          <div>
            <div class="gux-field">
              <input type="text" value="0">
            </div>
          </div>
        </gux-text-field>
      </div>
      <div>
        of 0
      </div>
    </div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;

exports[`gux-pagination-item-counts #render should render as expected (2) 1`] = `
<gux-pagination-buttons current-page="1" layout="full" total-pages="10">
  <div class="gux-full gux-pagination-buttons-container">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-input-container">
      <div>
        Page
      </div>
      <div class="gux-pagination-buttons-input">
        <gux-text-field use-clear-button="false" value="1">
          <div>
            <div class="gux-field">
              <input type="text" value="1">
            </div>
          </div>
        </gux-text-field>
      </div>
      <div>
        of 10
      </div>
    </div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;

exports[`gux-pagination-item-counts #render should render as expected (3) 1`] = `
<gux-pagination-buttons current-page="5" layout="full" total-pages="10">
  <div class="gux-full gux-pagination-buttons-container">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-input-container">
      <div>
        Page
      </div>
      <div class="gux-pagination-buttons-input">
        <gux-text-field use-clear-button="false" value="5">
          <div>
            <div class="gux-field">
              <input type="text" value="5">
            </div>
          </div>
        </gux-text-field>
      </div>
      <div>
        of 10
      </div>
    </div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;

exports[`gux-pagination-item-counts #render should render as expected (4) 1`] = `
<gux-pagination-buttons current-page="0" layout="small" total-pages="0">
  <div class="gux-pagination-buttons-container gux-small">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-spacer"></div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;

exports[`gux-pagination-item-counts #render should render as expected (5) 1`] = `
<gux-pagination-buttons current-page="1" layout="small" total-pages="10">
  <div class="gux-pagination-buttons-container gux-small">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-spacer"></div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;

exports[`gux-pagination-item-counts #render should render as expected (6) 1`] = `
<gux-pagination-buttons current-page="5" layout="small" total-pages="10">
  <div class="gux-pagination-buttons-container gux-small">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-spacer"></div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;

exports[`gux-pagination-item-counts #render should render as expected (7) 1`] = `
<gux-pagination-buttons current-page="0" layout="expanded" total-pages="0">
  <div class="gux-expanded gux-pagination-buttons-container">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-list-container"></div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;

exports[`gux-pagination-item-counts #render should render as expected (8) 1`] = `
<gux-pagination-buttons current-page="1" layout="expanded" total-pages="10">
  <div class="gux-expanded gux-pagination-buttons-container">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" disabled="" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-list-container">
      <button class="gux-pagination-buttons-list-current">
        1
      </button>
      <button class="gux-pagination-buttons-list-target">
        2
      </button>
      <button class="gux-pagination-buttons-list-target">
        3
      </button>
      <button class="gux-pagination-buttons-list-target">
        4
      </button>
      <button class="gux-pagination-buttons-list-target">
        5
      </button>
      <button class="gux-pagination-buttons-list-target">
        6
      </button>
      <button class="gux-pagination-buttons-list-target">
        7
      </button>
      <button class="gux-pagination-buttons-list-target">
        8
      </button>
      <button class="gux-pagination-buttons-list-target">
        9
      </button>
      <button class="gux-pagination-buttons-list-target">
        10
      </button>
    </div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;

exports[`gux-pagination-item-counts #render should render as expected (9) 1`] = `
<gux-pagination-buttons current-page="5" layout="expanded" total-pages="10">
  <div class="gux-expanded gux-pagination-buttons-container">
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="First" type="button">
          <gux-icon decorative="" iconname="ic-arrow-left-dbl"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Previous" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-left"></gux-icon>
        </button>
      </gux-button>
    </div>
    <div class="gux-pagination-buttons-list-container">
      <button class="gux-pagination-buttons-list-target">
        1
      </button>
      <button class="gux-pagination-buttons-list-target">
        2
      </button>
      <button class="gux-pagination-buttons-list-target">
        3
      </button>
      <button class="gux-pagination-buttons-list-target">
        4
      </button>
      <button class="gux-pagination-buttons-list-current">
        5
      </button>
      <button class="gux-pagination-buttons-list-target">
        6
      </button>
      <button class="gux-pagination-buttons-list-target">
        7
      </button>
      <button class="gux-pagination-buttons-list-target">
        8
      </button>
      <button class="gux-pagination-buttons-list-target">
        9
      </button>
      <button class="gux-pagination-buttons-list-target">
        10
      </button>
    </div>
    <div class="gux-pagination-buttons-group">
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Next" type="button">
          <gux-icon decorative="" iconname="ic-chevron-small-right"></gux-icon>
        </button>
      </gux-button>
      <gux-button>
        <!---->
        <button class="gux-secondary" title="Last" type="button">
          <gux-icon decorative="" iconname="ic-arrow-right-dbl"></gux-icon>
        </button>
      </gux-button>
    </div>
  </div>
</gux-pagination-buttons>
`;
